### PR TITLE
Added filter and search bar for log page

### PR DESCRIPTION
**Pull request by anishap16**
Added a filter bar with a search bar and filters. When the apply filters button is pressed, the filtered outfits are stored in a variable that is return by the function.

### CODE REVIEW

File: src/components/LogViewComponent/LogViewComponent.css [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+.outfit-item {
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+    margin-bottom: 1rem;
+    padding: 0.5rem;
+    border-radius: 5px;
+}
+
+.expand-btn {
```
Comment (sophia-hubscher) [inline src/components/LogViewComponent/LogViewComponent.css:31]: A lot of this CSS is never used in your file. I think you may have taken code from another file and forgotten to remove extra CSS?

File: src/components/LogViewComponent/LogViewComponent.js [hunk 2]
```diff
@@ -6,23 +7,209 @@ export class LogViewComponent extends BaseComponent {
...
+    // Search bar
+    const searchDiv = document.createElement("div");
+    const searchLabel = document.createElement("label");
+    searchLabel.htmlFor = "log-search";
+    searchLabel.textContent = "Search:";
+
+    const searchInput = document.createElement("input");
+    searchInput.type = "text";
+    searchInput.id = "log-search";
+    searchInput.placeholder = "Search by outfit ID, wardrobe item IDs, or notes";
```
Comment (sophia-hubscher) [inline src/components/LogViewComponent/LogViewComponent.js:58]: I would change this placeholder since these IDs are just used in the database and not visible to users. Honestly, I think you could get away with just removing the placeholder entirely if you want

File: src/components/LogViewComponent/LogViewComponent.js [hunk 3]
```diff
@@ -6,23 +7,209 @@ export class LogViewComponent extends BaseComponent {
...
-    const text = document.createElement('p');
-    text.textContent = 'Text here!';
+    // Create the filter bar
+    const filterBar = this.createFilterBar(outfits);
+    logContainer.appendChild(filterBar);
 
-    this.#container.appendChild(title);
-    this.#container.appendChild(text);
+    // Create the outfit list
+    const outfitListDiv = document.createElement("div");
```
Comment (sophia-hubscher) [inline src/components/LogViewComponent/LogViewComponent.js:31]: You defined CSS for this but didn't add it. I'd add the line `outfitListDiv.classList.add("outfit-list");`

File: src/components/LogViewComponent/LogViewComponent.js [hunk 4]
```diff
@@ -6,23 +7,209 @@ export class LogViewComponent extends BaseComponent {
...
+  createFilterBar(outfits) {
+    const filterBar = document.createElement("div");
+    filterBar.classList.add("log-filter-bar");
+
+    const filterTitle = document.createElement("h2");
+    filterTitle.textContent = "Filters";
+    filterBar.appendChild(filterTitle);
+
+    // Search bar
+    const searchDiv = document.createElement("div");
```
Comment (sophia-hubscher) [inline src/components/LogViewComponent/LogViewComponent.js:50]: This is really picky, but if you want the search bar to stretch across the whole filter bar, you could add an ID like "search-div" to this div and then add this CSS

```
#search-div {
    display: flex;
    align-items: center;
    width: 100%;
    gap: 5px;
}

#log-search {
    flex: 1;
}
```

File: src/components/LogViewComponent/LogViewComponent.js [hunk 5]
```diff
@@ -1,3 +1,4 @@
+import { getTestOutfits } from '../../testing/TestData.js';
```
Comment (sophia-hubscher) [inline src/components/LogViewComponent/LogViewComponent.js:1]: This should eventually be switched to IndexedDB, but that can totally be in another PR. I think that's what most people (myself included) have done so far

**Review by sophia-hubscher — CHANGES_REQUESTED** [review]
Overall this is looking great! I would just delete the unused CSS and maybe make those other small changes I suggested. On the whole though, this works perfectly

File: src/components/LogViewComponent/LogViewComponent.css (hunk 1 above)
Comment (anishap16) [reply src/components/LogViewComponent/LogViewComponent.css:31]: Yes! I used your CSS in order to keep the same styling across pages and I left the extra CSS there for whoever works on the rest of the page to use but I can remove it!

File: src/components/LogViewComponent/LogViewComponent.js (hunk 2 above)
Comment (anishap16) [reply src/components/LogViewComponent/LogViewComponent.js:58]: Done!

File: src/components/LogViewComponent/LogViewComponent.js (hunk 4 above)
Comment (anishap16) [reply src/components/LogViewComponent/LogViewComponent.js:50]: Added it, thank you!

File: src/components/LogViewComponent/LogViewComponent.css [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+.search-div {
+    display: flex;
+    align-items: center;
+    width: 100%;
+    gap: 5px;
+}
+
+.log-search {
```
Comment (sophia-hubscher) [inline src/components/LogViewComponent/LogViewComponent.css:35]: `log-search` is an ID not a class, so you need to use `#`

**Review by sophia-hubscher — CHANGES_REQUESTED** [review]

**Review by sophia-hubscher — APPROVED** [review]
Looks great!
